internal/builder/builder.go

```diff
@@ -148,11 +148,45 @@ func calculateMTKVariable(name string, build Builder, vars []variables.LagoonEnv
 
 // imagePatternParser parses the image pattern
 func imagePatternParser(pattern string, build Builder) string {
-	pattern = strings.Replace(pattern, "${database}", build.MTK.Database, 1)
+	var clean_database string
+	clean_database = replaceDoubleSpecial(build.MTK.Database)
+	pattern = strings.Replace(pattern, "${database}", clean_database, 1)
 	pattern = strings.Replace(pattern, "${service}", build.DockerComposeServiceName, 1)
 	pattern = strings.Replace(pattern, "${registry}", build.RegistryHost, 1)
 	pattern = strings.Replace(pattern, "${organization}", build.RegistryOrganization, 1)
 	pattern = strings.Replace(pattern, "${project}", variables.GetEnv("LAGOON_PROJECT", ""), 1)
 	pattern = strings.Replace(pattern, "${environment}", variables.GetEnv("LAGOON_ENVIRONMENT", ""), 1)
 	return pattern
 }
+
+// Replaces two of the same special character in a row with a single instance, because otherwise DockerHub will reject it
+// Inspired by https://stackoverflow.com/questions/59442559/how-to-compare-a-character-with-the-next-one-in-the-same-string 
+func replaceDoubleSpecial(source string) string {
+	var destination string
+	var lastSpecial bool
+	lastSpecial = false
```

Comment:
You can inline define
```suggestion
	lastSpecial := false
```